decryptor/postgresql/pg_general_decryptor.go

```diff
@@ -476,15 +397,176 @@ func (decryptor *PgDecryptor) MatchZoneInBlock(block []byte) {
 
 // GetTagBeginLength returns begin tag length, depends on decryptor type
 func (decryptor *PgDecryptor) GetTagBeginLength() int {
-	return decryptor.pgDecryptor.GetTagBeginLength()
+	return decryptor.binaryDecryptor.GetTagBeginLength()
 }
 
 // GetZoneIDLength returns begin tag length, depends on decryptor type
 func (decryptor *PgDecryptor) GetZoneIDLength() int {
-	return decryptor.pgDecryptor.GetTagBeginLength()
+	return decryptor.binaryDecryptor.GetTagBeginLength()
 }
 
 // SetDataProcessor replace current with new processor
 func (decryptor *PgDecryptor) SetDataProcessor(processor base.DataProcessor) {
 	decryptor.dataProcessor = processor
 }
+
+// OnColumn handler which process column data on db response and try to decrypt/detect poison record
+func (decryptor *PgDecryptor) OnColumn(ctx context.Context, data []byte) (context.Context, []byte, error) {
+	logger := logging.GetLoggerFromContext(ctx)
+	span := trace.FromContext(ctx)
+	// try to skip small piece of data that can't be valuable for us
+	if (decryptor.IsWithZone() && len(data) >= zone.ZoneIDBlockLength) || len(data) >= base.GetMinAcraStructLength() {
+		decryptor.Reset()
+		span.AddAttributes(trace.BoolAttribute("decryption", true))
+
+		// Zone anyway should be passed as whole block
+		// so try to match before any operations if we process with ZoneMode on
+		if decryptor.IsWithZone() && !decryptor.IsMatchedZone() {
+			span.AddAttributes(trace.BoolAttribute("match_zone", true))
+			// try to match zone
+			decryptor.MatchZoneBlock(data)
+			var err error
+			if decryptor.IsWholeMatch() {
+				// check that it's not poison record
+				err = checkWholePoisonRecord(data, decryptor, logger)
+			} else {
+				// check that it's not poison record
+				err = checkInlinePoisonRecordInBlock(data, decryptor, logger)
+			}
+
+			if err != nil {
+				logger.WithField(logging.FieldKeyEventCode, logging.EventCodeErrorDecryptorCantCheckPoisonRecord).WithError(err).Errorln("Can't check poison record in block")
+				return ctx, nil, err
+			}
+			// TODO remove to back decryption only with matched zone_id
+			//return ctx, data, nil
+		}
+
+		// create temporary logger to log related zone_id in flow of decryption
+		// client_id set on higher level on start of processing connection
+		decryptionLogger := logger.WithField("zone_id", string(decryptor.GetMatchedZoneID()))
+		var newData []byte
+		var err error
+		if decryptor.IsWholeMatch() {
+			newData, err = decryptor.processWholeBlockDecryption(ctx, data, decryptionLogger)
+			if err != nil {
+				decryptionLogger.WithField(logging.FieldKeyEventCode, logging.EventCodeErrorDecryptorCantDecryptBinary).WithError(err).Errorln("Can't process whole block")
+				return ctx, nil, err
+			}
+			return ctx, newData, err
+		}
+		newData, err = decryptor.processInlineBlockDecryption(ctx, data, decryptionLogger)
```

Comment:
```suggestion
                // else "injected cell mode", inline acrastruct
		newData, err = decryptor.processInlineBlockDecryption(ctx, data, decryptionLogger)
```

Reply:
will add in commit to avoid merging after commiting from github